addons/product_images/models/ir_cron_trigger.py

```diff
@@ -0,0 +1,26 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import _, api, models
+from odoo.exceptions import ValidationError
+
+
+class ir_cron_trigger(models.Model):
+    _inherit = "ir.cron.trigger"
+
+    @api.constrains('cron_id')
+    def _check_cron_id(self):
+        """ Check if `ir_cron_fetch_image` is already triggered
+
+        Ensure that there is a maximum of one trigger at a time for `ir_cron_fetch_image`.
+
+        :return: None
+        :raises ValidationError: If there is already an `ir_cron_fetch_image` triggered
+        """
+        ir_cron_fetch_image_id = self.env.ref('product_images.ir_cron_fetch_image').id
```

Comment:
```suggestion
        ir_cron_fetch_image_id = self.env.ref(
            'product_images.ir_cron_fetch_image', raise_if_not_found=False
        ).id
```
If the admin deletes the cron you want an error to be raised when he tries to fetch some images, not when any other module triggers a cron.

Reply:
Wrong suggestion, in case the cron is missing (it shoud not) `ref()` returns `None` but `None` as no `id` attribute. The cron should not be removed, instead the admin can archive it if he wants to disable it.

Reply:
`.id` should be moved in another line, indeed.

> The cron should not be removed, instead the admin can archive it if he wants to disable it.

Do not underestimate user mistakes, protecting against it (unless it requires overly complex code) is always a good idea, even more for core methods.

Reply:
Sure, the suggestion can be improved. We can work with `ir_cron_fetch_image` and check if it exists first, for example. But the, we must either prevent the user from deleting the cron or assume that they can delete the cron.

Reply:
Re-reading the class I would actually `raise_if_not_found=False`, just pay attention to test for `ir_cron_fetch_image is not None` and

Reply:
I've added this UserError if the cron has been deleted but since it should only be accessible by an administrator, I finally think that my message is a bit off. What do you think?

https://github.com/odoo/odoo/blob/2a79943ffb20e8df973237955eee7db56e263ff8/addons/product_images/wizard/product_fetch_image_wizard.py#L54-L57

Reply:
I would emit a warning at worst, not fail with an exception. Here you would raise the exception whenever you create/write a cron, even if it is not related to yours

Reply:
This is a user error from the wizard, not from `ir_cron_trigger.py`. I should have made it more clear for you, sorry for that.